modules/internal/localeModels/flat/LocaleModel.chpl

```diff
@@ -286,32 +286,44 @@ module LocaleModel {
   private extern proc chpl_memhook_md_num(): chpl_mem_descInt_t;
 
   // The allocator pragma is used by scalar replacement.
+
+  // Note that there's 2 chpl_here_alloc() procs. This one explicitly takes an
+  // int(64) and is marked with "locale model alloc" while the one below takes
+  // a generic `integral`. This one is used/called by the compiler and so its
+  // type must be fully defined.
   pragma "allocator"
   pragma "locale model alloc"
```

Comment:
Each locale model can only have a single `locale model alloc` (and actually this code could probably be moved to the parent class.) Nothing enforces that there can only be one, but only one will be used by the compiler as it will override any previous definition with the latest one it sees with [this code] (https://github.com/chapel-lang/chapel/blob/master/compiler/passes/normalize.cpp#L114)

`chpl_here_alloc()` used to be fully generic, but it was changed to take int(64) in https://github.com/chapel-lang/chapel/commit/b20e1b2ba00f2d6cda008f7dc2b205cbd2e160d3 in order to support having the compiler directly call it. A comment that that commit removed mentioned: 

```c
// For now we produce a PRIM_CHPL_ALLOC here.  Eventually it would be
// nice to directly produce a call to chpl_here_alloc().  Currently we
// can't do that because chpl_here_alloc() is generic in the types of
// its first argument and its return, and some of the calls to this
// function occur after resolution.  In addition, in several places in
// the compiler we look specifically for PRIM_CHPL_ALLOC in the AST,
// and we would presumably need to be able to find the equivalent AST
// nodes even if we switched to emitting direct calls.
```